### PR TITLE
add Dash network

### PR DESCRIPTION
**Pull request by nmarley**
Enables library to be used for Dash projects as well.

### CODE REVIEW

File: tests/Network/NetworkTest.php [hunk 1]
```diff
@@ -200,6 +200,28 @@ public function testFactoryPresets()
...
+        $this->assertEquals(NetworkFactory::dash()->getHDPubByte(), '02fe52f8');
+        $this->assertEquals(NetworkFactory::dash()->getNetMagicBytes(), 'bd6b0cbf');
+        $this->assertEquals('7X7VPCbTMLvUSjhMo3vdqKb8eNrccxgkJ1', $p2sh->getAddress(NetworkFactory::dash()));
+        $this->assertEquals('XbKZStn8KGzRUsSr5wiq18A3VUyD7pdKXX', $p2pk->getAddress(NetworkFactory::dash()));
+
+        // Dash testnet
+        $this->assertEquals(NetworkFactory::dashTestnet()->getAddressByte(), '8b');
+        $this->assertEquals(NetworkFactory::dashTestnet()->getP2shByte(), '13');
+        $this->assertEquals(NetworkFactory::dashTestnet()->getPrivByte(), 'ef');
+        $this->assertEquals(NetworkFactory::dashTestnet()->isTestnet(), true);
```
Comment (afk11) [inline tests/Network/NetworkTest.php]: https://travis-ci.org/Bit-Wasp/bitcoin-php/builds/105893301

Get this test working by adding a 4th param `true` in the `NetworkFactory::dashTestnet()`

**Comment by afk11** [comment]
Great stuff, thanks! Can you check the above, you indicate that it's testnet by passing $bool, which defaults to false.

**Comment by afk11** [comment]
#235 was merged, I'll tag a release once this goes in!

**Comment by nmarley** [comment]
Ah, ok, sorry about that! Flagged Dash and Startcoin testnets correctly in 5e6f3e0.

**Comment by afk11** [comment]
Merged, thanks! `v0.0.19.4`

**Comment by murich** [comment]
```
$network = \BitWasp\Bitcoin\Network\NetworkFactory::dash();
        $master = HierarchicalKeyFactory::fromExtended(
            "xpub68NZH1dmpQzLjmSkh1boBTzYoETTouKdsFK2eQRZXzVtqY9Emto6jmPJqRyuvvxPUTNbud4VmMfcRSR7FkCT6PNsGq6bwBPGrqE9bpyy4ps",
            $network
        );
```

Results in
```
InvalidArgumentException [ 0 ]: HD key magic bytes do not match network magic bytes 
```

I got this xpub from https://iancoleman.io/bip39/#english

What could be potentially wrong here?

**Comment by nmarley** [comment]
> What could be potentially wrong here?

Looks like the HDKey version bytes weren't updated after the last change ~16 months ago:

https://github.com/dashpay/dash/pull/926

**Pull request merged**